src/DGtal/geometry/tools/Hull2DHelpers.h

```diff
@@ -0,0 +1,375 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file Hull2DHelpers.h
+ * @author Tristan Roussillon (\c tristan.roussillon@liris.cnrs.fr )
+ * Laboratoire d'InfoRmatique en Image et Systèmes d'information - LIRIS (CNRS, UMR 5205), CNRS, France
+ *
+ * @date 2013/12/02
+ *
+ * Header file for module Hull2DHelpers.cpp
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(Hull2DHelpers_RECURSES)
+#error Recursive header files inclusion detected in Hull2DHelpers.h
+#else // defined(Hull2DHelpers_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define Hull2DHelpers_RECURSES
+
+#if !defined Hull2DHelpers_h
+/** Prevents repeated inclusion of headers. */
+#define Hull2DHelpers_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <list>
+#include <vector>
+#include "boost/utility.hpp"
+
+#include "DGtal/base/Common.h"
+#include "DGtal/base/IteratorCirculatorTraits.h"
+#include "DGtal/base/FrontInsertionSequenceToStackAdapter.h"
+#include "DGtal/base/BackInsertionSequenceToStackAdapter.h"
+#include "DGtal/base/CStack.h"
+#include "DGtal/geometry/tools/CPolarPointComparator2D.h"
+#include "DGtal/geometry/tools/PolarPointComparatorBy2x2DetComputer.h"
+#include "DGtal/geometry/tools/determinant/COrientationFunctor2.h"
+#include "DGtal/geometry/tools/determinant/PredicateFromOrientationFunctor2.h"
+#include "DGtal/geometry/tools/determinant/AvnaimEtAl2x2DetSignComputer.h"
+#include "DGtal/geometry/tools/determinant/Simple2x2DetComputer.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  namespace functions
+  {
+  /**
+   * @namespace Hull2D gathers useful functions to compute 
+   * and return the convex hull or the alpha-shape of a range 
+   * of 2D points. 
+   */
+  namespace Hull2D
```

Comment:
This is important for DGtal design: shall we allow to have inner, topic dependent,  namespaces ?
So far I don't know